js/runner.go

```diff
@@ -318,9 +321,55 @@ func (r *Runner) SetOptions(opts lib.Options) error {
 		r.console = c
 	}
 
+	// FIXME: Resolver probably shouldn't be reset here...
+	// It's done because the js.Runner is created before the full
+	// configuration has been processed, at which point we don't have
+	// access to the DNSConfig, and need to wait for this SetOptions
+	// call that happens after all config has been assembled.
+	// We could make DNSConfig part of RuntimeOptions, but that seems
+	// conceptually wrong since the JS runtime doesn't care about it
+	// (it needs the actual resolver, not the config), and it would
+	// require an additional field on Bundle to pass the config through,
+	// which is arguably worse than this.
+	ttl, err := parseTTL(opts.DNS.TTL.String)
+	if err != nil {
+		return err
+	}
+	strategy := opts.DNS.Strategy.DNSStrategy
+	if !strategy.IsADNSStrategy() {
+		strategy = lib.DefaultDNSConfig().Strategy.DNSStrategy
+	}
+	r.Resolver = netext.NewResolver(r.ActualResolver, ttl, strategy)
+
 	return nil
 }
 
+func parseTTL(ttlS string) (time.Duration, error) {
+	ttl := time.Duration(0)
+	switch ttlS {
+	case "inf":
+		// cache "indefinitely"
```

Comment:
😄 I purposely used "indefinitely" since it implies a duration/time, whereas "infinitely" can be used in other contexts. Even though "inf" is a common constant used for this purpose, and makes sense here. And the quotes are because it's not really infinite/indefinite, just a long duration.

Anyway, sure, if you think it's confusing I'll change it to `"infinitely"`, although at that point the comment is kind of redundant. :)